srtcore/buffer.cpp

```diff
@@ -63,6 +63,41 @@ using namespace std;
 using namespace srt_logging;
 using namespace srt::sync;
 
+bool AvgBufSize::canUpdate(const time_point& now) const
+{
+    const int      SRT_MAVG_BASE_PERIOD = 1000000; // us
+    const int      SRT_us2ms            = 1000;
+    const uint64_t elapsed_ms           = count_milliseconds(now - m_tsLastSamplingTime); // ms since last sampling
+    return (elapsed_ms >= (SRT_MAVG_BASE_PERIOD / SRT_MAVG_SAMPLING_RATE) / SRT_us2ms);
```

Comment:
Wouldn't it be simpler if you use comparisong against `count_microseconds(SRT_MAVG_BASE_PERIOD / SRT_MAVG_SAMPLING_RATE)` ? If the internal representation is in microseconds, this wouldn't do any calculations, whereas `count_milliseconds` will definitely do.

Reply:
Here we don't use `sync::duration`. I am not sure if it is worth to use it due to potential performance penalties, at least in C++11 version.

Reply:
At least for clarity you could add a new constant for `MAVG_PERIOD_MS` or something.

Reply:
Added.